lib/_emerge/EbuildFetcher.py

```diff
@@ -273,17 +281,43 @@ def _target(settings, manifest, uri_map, fetchonly):
         if _want_userfetch(settings):
             _drop_privs_userfetch(settings)
 
-        rval = 1
         allow_missing = manifest.allow_missing or "digest" in settings.features
-        if fetch(
-            uri_map,
-            settings,
-            fetchonly=fetchonly,
-            digests=copy.deepcopy(manifest.getTypeDigests("DIST")),
-            allow_missing_digests=allow_missing,
-        ):
-            rval = os.EX_OK
-        return rval
+
+        async def main():
+            loop = asyncio.get_event_loop()
+            task = asyncio.ensure_future(
+                async_fetch(
+                    uri_map,
+                    settings,
+                    fetchonly=fetchonly,
+                    digests=copy.deepcopy(manifest.getTypeDigests("DIST")),
+                    allow_missing_digests=allow_missing,
+                )
+            )
+
+            def sigterm_handler(signum, _frame):
+                loop.call_soon_threadsafe(task.cancel)
+                signal.signal(signal.SIGTERM, signal.SIG_IGN)
+
+            signal.signal(signal.SIGTERM, sigterm_handler)
+            try:
+                await task
+            except asyncio.CancelledError:
+                # If asyncio.CancelledError arrives too soon after fork/spawn
+                # then handers will not have an opportunity to termintate
+                # the corresponding process, so clean up after this race.
+                for proc in multiprocessing.active_children():
+                    proc.terminate()
+                    proc.join(0)
+                    if proc.exitcode is None:
+                        proc.kill()
+                        proc.join(0)
```

Comment:
That `proc.join(0)` right after kill seems unnecessary. I think it should be either dropped, or invoked without any argument.

Reply:
The intention is to try and ensure that the pid is removed from the process table as soon as possible, since the unit test fails otherwise, and this can only happen if the process waited upon (either by its parent or a pid 1 reaper).

We should be able to omit the argument, since SIGKILL should be completely effective here.

Reply:
Argument removed.